src/mmap_unix.rs

```diff
@@ -74,6 +74,142 @@ impl error::Error for Error {}
 
 pub type Result<T> = result::Result<T, Error>;
 
+/// A factory struct to build `MmapRegion` objects.
+pub struct MmapRegionBuilder<B = ()> {
+    size: usize,
+    prot: i32,
+    flags: i32,
+    file_offset: Option<FileOffset>,
+    raw_ptr: Option<*mut u8>,
+    hugetlbfs: Option<bool>,
+    bitmap: B,
+}
+
+impl<B: Bitmap> MmapRegionBuilder<B> {
+    /// Create a new `MmapRegionBuilder`.
+    pub fn new(size: usize, bitmap: B) -> Self {
```

Comment:
how about renaming "new()" to "new_with_bitmap()" and add:
```
impl<B: Bitmap + Default> MmapRegionBuilder<B> {
         pub fn new(size: usize) -> Self {
               Self::new_with_bitmap(size, B::default)
         }
}

Reply:
Cool, done.